nixos/default.nix

```diff
@@ -129,9 +129,19 @@
         default = null;
       };
 
+      port = mkOption {
+        description = "Port to listen for HTTP requests on";
+        type with types; nullOr port;
+        default = null;
+      };
+
       toURL = mkOption {
         description = "URL to forward HTTP requests to";
         type = types.str;
+        default =
+          if defaults.url != null && config.port != null
+          then "${defaults.url}:${builtins.toString config.port}"
```

Comment:
Once we have urlParts.* in the service, the URL here can be constructed from those parts.